frontend/app/components/input-select.tsx

```diff
@@ -0,0 +1,79 @@
+import { type ComponentProps, type ReactNode, forwardRef } from 'react';
+
+import clsx from 'clsx';
+
+import { InputError } from './input-error';
+import { InputHelp } from './input-help';
+import { InputLabel } from './input-label';
+import { InputOption } from './input-option';
+
+export interface InputSelectProps extends Omit<ComponentProps<'select'>, 'aria-describedby' | 'aria-errormessage' | 'aria-invalid' | 'aria-labelledby' | 'aria-required'> {
+  errorMessage?: string;
+  helpMessage?: ReactNode;
+  helpMessageSecondary?: ReactNode;
+  id: string;
+  label: string;
+  name: string;
+  options: ComponentProps<typeof InputOption>[];
+}
+
+const InputSelect = forwardRef<HTMLSelectElement, InputSelectProps>((props, ref) => {
+  const { errorMessage, helpMessage, helpMessageSecondary, id, label, options, className, required, ...restInputProps } = props;
+
+  const inputErrorId = `input-${id}-error`;
+  const inputHelpMessageId = `input-${id}-help`;
+  const inputHelpMessageSecondaryId = `input-${id}-help-secondary`;
+  const inputLabelId = `input-${id}-label`;
+  const inputWrapperId = `input-${id}`;
+
+  function getAriaDescribedby() {
+    const ariaDescribedby = [];
+    if (helpMessage) ariaDescribedby.push(inputHelpMessageId);
+    if (helpMessageSecondary) ariaDescribedby.push(inputHelpMessageSecondaryId);
+    return ariaDescribedby.length > 0 ? ariaDescribedby.join(' ') : undefined;
+  }
+
+  return (
+    <div id={inputWrapperId} data-testid={inputWrapperId} className="form-group">
+      <InputLabel id={inputLabelId} htmlFor={id} required={required}>
+        {label}
+      </InputLabel>
+      {errorMessage && (
+        <InputError id={inputErrorId} className="mb-1.5">
+          {errorMessage}
+        </InputError>
+      )}
+      {helpMessage && (
+        <InputHelp id={inputHelpMessageId} className="mb-1.5" data-testid="input-select-help">
+          {helpMessage}
+        </InputHelp>
+      )}
+      <select
+        ref={ref}
+        aria-describedby={getAriaDescribedby()}
+        aria-errormessage={errorMessage && inputErrorId}
+        aria-invalid={!!errorMessage}
+        aria-labelledby={inputLabelId}
+        aria-required={required}
+        className={clsx('form-control', className)}
+        data-testid="input-select"
+        id={id}
+        required={required}
+        {...restInputProps}
+      >
+        {options.map(({ label, value, ...restProps }) => (
+          <InputOption label={label} value={value} key={value} {...restProps} />
```

Comment:
Pass the id into `<InputOption>` so it can be mapped to the testid for the option element.

Reply:
```suggestion
          <InputOption key={value} {...restProps} />
```

Reply:
@sebastien-comeau Since `label` and `value` are currently mandatory (as defined by `InputOptionProps`), we have to explicitly set them when using `<InputOption />`. Thoughts?

Reply:
> @sebastien-comeau Since `label` and `value` are currently mandatory (as defined by `InputOptionProps`), we have to explicitly set them when using `<InputOption />`. Thoughts?

Correction: I had de-structured  `label` and `value` from `InputOptionProps`, meaning `label` was missing from `...restProps`.

`<InputOption>` now accepts an `id`, which is used to construct a value for `data-testId`.